survey/survey-api/src/main/java/com/duckduckgo/survey/api/SurveyParameterManager.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.survey.api
+
+interface SurveyParameterManager {
+    suspend fun canProvideAllParameters(requestedQueryParams: List<String>): Boolean
```

Comment:
Why do we need this method?
I'd also add javadocs to public api methods.

Reply:
Updated!